mongodbatlas/provider.go

```diff
@@ -303,35 +307,101 @@ func addBetaFeatures(provider *schema.Provider) {
 	}
 }
 
+func getValueFromConfigOrEnv(d *schema.ResourceData, attrName string, envVars []string) string {
+	var val = d.Get(attrName).(string)
+	if val == "" {
+		val = multiEnvDefaultFunc(envVars, "").(string)
+	}
+	return val
+}
+
 func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	var (
+		publicKey    string
+		privateKey   string
+		realmBaseURL string
+	)
 	mongodbgovCloud := pointy.Bool(d.Get("is_mongodbgov_cloud").(bool))
 	if *mongodbgovCloud {
 		baseURL = "https://cloud.mongodbgov.com"
 	} else {
-		baseURL = d.Get("base_url").(string)
-	}
+		baseURL = getValueFromConfigOrEnv(d, "base_url", []string{
+			"MONGODB_ATLAS_BASE_URL",
+			"MCLI_OPS_MANAGER_URL",
+		})
+	}
+
+	publicKey = getValueFromConfigOrEnv(d, "public_key", []string{
+		"MONGODB_ATLAS_PUBLIC_KEY",
+		"MCLI_PUBLIC_API_KEY",
+	})
+	if publicKey == "" {
+		return nil, diag.FromErr(fmt.Errorf(AttrNotSetError, "public_key"))
+	}
+
+	privateKey = getValueFromConfigOrEnv(d, "private_key", []string{
+		"MONGODB_ATLAS_PRIVATE_KEY",
+		"MCLI_PRIVATE_API_KEY",
+	})
+	if privateKey == "" {
+		return nil, diag.FromErr(fmt.Errorf(AttrNotSetError, "private_key"))
+	}
+
+	realmBaseURL = getValueFromConfigOrEnv(d, "realm_base_url", []string{
+		"MONGODB_REALM_BASE_URL",
+	})
+
+	// Commenting out AssumeRole/AWS attributes for the purpose of migrating Project resource as part of POC
```

Comment:
Intentionally left out AssumeRole migration in the provider for now for the purpose of POC